src/components/network-chart/NetworkChartRow.js

```diff
@@ -29,117 +30,109 @@ import type { NetworkPayload } from '../../types/markers';
-  const { marker, networkPayload } = props;
+  const { marker, networkPayload, markerWidth } = props;
   const { start, dur } = marker;
 
-  // A marker does not always contain the same set of networkPayload on the start of
-  // the connection.
-  const queueStart =
-    networkPayload.secureConnectionStart ||
-    networkPayload.tcpConnectEnd ||
-    networkPayload.connectStart ||
-    networkPayload.domainLookupStart ||
-    0;
-
-  // The default for width of the phases is always zero.
-  let requestQueue: number = 0;
-  let request: number = 0;
-  let response: number = 0;
-
-  // Timestamps from network markers aren't adjusted when processing
-  // the gecko profile format. This is the reason why the start timestamp of a
-  // marker can be after the start timestamp of the request.
-  // To prevent false networkPayload visualization, we need this workaround.
-  // See https://github.com/firefox-devtools/profiler/issues/1493 for more detail.
-  if (
-    networkPayload.requestStart &&
-    start < networkPayload.requestStart &&
-    networkPayload.responseStart &&
-    start < networkPayload.responseStart
-  ) {
-    request =
-      (networkPayload.responseStart - networkPayload.requestStart) / dur * 100;
-  }
-
-  if (
-    networkPayload.responseEnd &&
-    start < networkPayload.responseEnd &&
-    networkPayload.responseStart &&
-    start < networkPayload.responseStart
-  ) {
-    response =
-      (networkPayload.responseEnd - networkPayload.responseStart) / dur * 100;
-  }
-
-  if (queueStart && start < queueStart && queueStart > 0) {
-    requestQueue = (queueStart - start) / dur * 100;
-  }
+  const barParts = [];
+  let previousValue = start;
+  let previousName = 'startTime';
 
-  // When we keep the default values (=zero), the
-  // response takes the full width of the bar to be more visible.
-  if (requestQueue + request + response === 0) {
-    response = 100;
-  }
+  PROPERTIES_IN_ORDER.filter(
+    property => typeof networkPayload[property] === 'number'
+  ).forEach((property, i, { length }) => {
+    // We force-coerce the value into a number just to appease Flow. Indeed the
+    // previous filter ensures that all values are numbers but Flow can't know
+    // that.
+    const value = +networkPayload[property];
+    barParts.push({
+      left: (previousValue - start) / dur * markerWidth,
+      width: Math.max((value - previousValue) / dur * markerWidth, 1),
+      opacity: i / (length - 1),
+      name: property,
+      previousName,
+      value,
+      duration: value - previousValue,
+    });
+    previousValue = value;
+    previousName = property;
+  });
 
-  // Adding either the default value (=zero) as width or
-  // the new, calculated value that can be added.
-  const requestQueueWidth = requestQueue;
-  const requestWidth = request;
-  const responseWidth = response;
+  // The last part isn't generally colored (opacity is 0) unless it's the only
+  // one, and in that case it covers the whole duration.
```

Comment:
Note: I decided to add this last part even when it's not visible so that screen reader users can still have it with the aria-label attribute for this element.